### PR TITLE
Make the homepage script more posixly

### PR DESCRIPTION
**Pull request by KellerFuchs**
- Fix quoting;
- avoid non-POSIX-sh features, like `echo` flags;
- fix all the `shellcheck` warnings;
- make `shellcheck` errors fail `test.sh`.

### CODE REVIEW

**Comment by KellerFuchs** [comment]
This currently breaks something, do **not** merge without testing.

File: src/hashbang.sh [hunk 1]
```diff
@@ -297,30 +297,29 @@ while true; do
 		break;
 	fi
 done
-host=$(cat $host_data | head -n $choice | tail -n1 | cut -d '|' -f1)
+host=$(head -n "$choice" | tail -n1 | cut -d '|' -f1 < "$host_data")
```
Comment (daurnimator) [inline src/hashbang.sh]: just pass the filename to `head`

Comment (KellerFuchs) [inline src/hashbang.sh]: Good point, but for now let's sleep for real

**Comment by necrophcodr** [comment]
All in all, this seems really really well done and pretty solid.

:+1:

**Comment by KellerFuchs** [comment]
@necrophcodr Not sure what your comments are reffering to.  Did you mean to use line comments?

**Comment by KellerFuchs** [comment]
Ok, this should be good now.

Someone else reviews, tests and confirms?  @necrophcodr ?

**Comment by KellerFuchs** [comment]
Just rebased on `master` so that we can merge once the CI is green and someone reviews.

**Comment by KellerFuchs** [comment]
```
16:59:32           +liam │ KellerFuchs: Also, #70 looks good to me. Why are you switching from Travis Containers
                         │ to trusty?
17:07:08     KellerFuchs │ liam: Missed the actually-reevant discussion in the flamewar: switched to trusty to
                         │ actually have a shellcheck package

**Pull request merged**